File: routes/securityQuestionRoutes.js
const express = require('express');
const router = express.Router();
const securityQuestionController = require('../controllers/securityQuestionController');

// Add or update security questions
router.post('/', securityQuestionController.addOrUpdateSecurityQuestions);

// Get all security questions for a user
router.get('/', securityQuestionController.getSecurityQuestions);

// Update a security question
router.put('/:id', securityQuestionController.updateSecurityQuestion);

// Delete a security question
router.delete('/:id', securityQuestionController.deleteSecurityQuestion);

module.exports = router;

File: routes/security_question_routes.py
from fastapi import APIRouter
from controllers.security_question_controller import SecurityQuestionController

router = APIRouter(prefix="/security-questions", tags=["Security Questions"])

@router.post("/", response_model=dict)
async def add_or_update_security_questions(data: dict):
    return await SecurityQuestionController.add_or_update_security_questions(data)

@router.get("/", response_model=list)
async def get_security_questions():
    return await SecurityQuestionController.get_security_questions()

@router.put("/{id}", response_model=dict)
async def update_security_question(id: str, data: dict):
    return await SecurityQuestionController.update_security_question(id, data)

@router.delete("/{id}", response_model=dict)
async def delete_security_question(id: str):
    return await SecurityQuestionController.delete_security_question(id)